src/Pages/Courses/CoursesPage.js

```diff
@@ -0,0 +1,90 @@
+import React, { useEffect, useState } from 'react';
+import { useHistory } from 'react-router-dom';
+import { Col, Container, Row } from 'reactstrap';
+import './courses-page.css';
+import CourseCard from './CourseCard';
+import Header from '../../Components/Header/Header';
+import { getAllCourses } from '../../APIFunctions/Courses';
+
+// Group together every 3 course cards
+export function groupCards(courses) {
+  let groups = [];
+  let temp = [];
+
+  courses.map((course, index) => {
+    if (index%3 === 0 && index !== 0) {
+      groups.push(temp);
+      temp = [];
+    }
+
+    temp.push(course);
+  });
+
+  if (temp.length !== 0) {
+    groups.push(temp);
+  }
+
+  return groups;
+}
+
+export default function CoursesList() {
+  const [coursesList, setCoursesList] = useState({});
+  const history = useHistory();
+  let cardNum = 0;
+
+
+  async function getCourses() {
+    const data = await getAllCourses();
+    setCoursesList(data.responseData);
+  }
+
+  useEffect(() => {
+    getCourses();
+  }, []);
+
+  const headerProp = {
+    title: 'SCE Courses Page'
+  };
+
+  function handleClick(clickedCourse) {
+    history.push(`/course/${clickedCourse.title}/lessons/summary`, {
```

Comment:
The redirection to /courses is something that is done on purpose because in LessonPage.js (lines 26-30), I check to see if the course id from mongodb was passed to LessonPage.js from CoursesPage.js. I need the course id since I use that to uniquely identify a course and update its lessons in the backend. If a user copies and pastes a lesson link in the browser and I don't do the check, then editing a lesson wouldn't work because the course id would be undefined. 

I can resolve this by finding the course name in mongodb and getting the id, but I would have to add a check when creating a course to ensure that all course names are unique.